File: ncPersistencia/pCrediario.vb
Imports ncDados.nsCrediario
Imports ncComum.nsAcessoBD
Imports ncComum.nsFuncoes
Imports ncComum.nsExcecao

Namespace nsCrediario

    Public Class pCrediario

        Public Function Listar() As ColecaoCrediario

            Dim retorno As ColecaoCrediario
            Dim acessoBanco As cAcessoBD
            Dim ds As DataSet
            Dim dt As DataTable
            Dim row As DataRow
            Dim item As dCrediario
            Dim comandoSQL As String


            Try

                acessoBanco = New cAcessoBD


                comandoSQL = " Select cid, controle, usuarioId, clienteId, " & _
                             "parcelas, valortotal, valorpago, saldodevedor, dataVenda From crediario"

                ds = acessoBanco.ExecutarDS(comandoSQL)

                If Not ds Is Nothing Then
                    If ds.Tables.Count > 0 Then
                        dt = ds.Tables(0)

                        If dt.Rows.Count > 0 Then
                            retorno = New ColecaoCrediario

                            For Each row In dt.Rows
                                item = New dCrediario

                                item.cid = cFuncoes.RetornarInteiro(row("cid"))
                                item.controle = cFuncoes.RetornarInteiro(row("controle"))
                                item.ValorTotal = cFuncoes.PersistirDecimal(row("valortotal"))
                                item.ValorTotal = cFuncoes.PersistirDecimal(row("valorpago"))
                                item.ValorTotal = cFuncoes.PersistirDecimal(row("saldodevedor"))
                                item.usuarioId = cFuncoes.RetornarInteiro(row("usuarioId"))
                                item.clienteId = cFuncoes.RetornarInteiro(row("clienteId"))
                                item.Parcelas = cFuncoes.PersistirInteiro(row("parcelas"))
                                item.DataVenda = cFuncoes.RetornarData(row("dataVenda"))

                                retorno.Add(item)
                            Next
                        Else
                            retorno = Nothing
                        End If
                    Else
                        retorno = Nothing
                    End If
                Else
                    retorno = Nothing
                End If

            Catch ex As Exception

                retorno = Nothing
                Throw New ExcecaoNascomercio("Erro em Listar Crediário [" & Me.ToString() & "] - " & ex.Message)

            End Try

            Listar = retorno

        End Function

        Public Function ConsultarMax() As Integer

            Dim acessoBanco As cAcessoBD
            Dim ds As DataSet
            Dim dt As DataTable
            Dim row As DataRow
            Dim sqlSelect As String
            Dim sqlFrom As String
            Dim retorno As Integer

            Try

                acessoBanco = New cAcessoBD


                sqlSelect = " Select MAX(controle) as controle"
                sqlFrom = " From crediario "

                ds = acessoBanco.ExecutarDS(sqlSelect & " " & sqlFrom)

                If Not ds Is Nothing Then
                    If ds.Tables.Count > 0 Then
                        dt = ds.Tables(0)

                        If dt.Rows.Count > 0 Then

                            For Each row In dt.Rows
                                retorno = IIf(row("controle") Is DBNull.Value, 0, cFuncoes.RetornarInteiro(row("controle")))
                            Next
                        Else
                            retorno = Nothing
                        End If
                    Else
                        retorno = Nothing
                    End If
                Else
                    retorno = Nothing
                End If

            Catch ex As Exception

                retorno = Nothing
                Throw New ExcecaoNascomercio("Erro em ConsultarMax Venda [" & Me.ToString() & "] - " & ex.Message)

            End Try

            Return retorno

        End Function

        Public Function Consultar(ByVal dados As dCrediario) As ColecaoCrediario

            Dim retorno As ColecaoCrediario
            Dim acessoBanco As cAcessoBD
            Dim ds As DataSet
            Dim dt As DataTable
            Dim row As DataRow
            Dim item As dCrediario
            Dim sqlSelect As String
            Dim sqlWhere As String
            Dim sqlFrom As String


            Try

                acessoBanco = New cAcessoBD


                sqlSelect = " Select cid, controle, usuarioId, clienteId, " & _
                             "parcelas, valortotal, valorpago, saldodevedor, dataVenda "

                sqlWhere = String.Empty
                sqlFrom = " From crediario "

                '-- cid
                If dados.cid <> 0 Then
                    sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.cid, "cid")
                End If

                '-- controle
                If dados.controle <> 0 Then
                    sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.controle, "controle")
                End If

                '-- usuarioId
                If dados.usuarioId <> 0 Then
                    sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.usuarioId, "usuarioId")
                End If

                '-- clienteId
                If dados.clienteId <> 0 Then
                    sqlWhere = cFuncoes.MontarParametrosSQL(sqlWhere, dados.clienteId, "clienteId")
                End If

                '-- saldo devedor
                If dados.SaldoDevedor > 0.001 Then
                    If Not sqlWhere.Equals(String.Empty) Then
                        sqlWhere += " AND saldodevedor > 0.001 "
                    Else
                        sqlWhere = " saldodevedor > 0.001 "
                    End If
                End If

                If Not sqlWhere.Equals(String.Empty) Then
                    sqlWhere = " WHERE " & sqlWhere
                End If

                ds = acessoBanco.ExecutarDS(sqlSelect & " " & sqlFrom & " " & sqlWhere & " order by year(dataVenda), month(dataVenda)")

                If Not ds Is Nothing Then
                    If ds.Tables.Count > 0 Then
                        dt = ds.Tables(0)

                        If dt.Rows.Count > 0 Then
                            retorno = New ColecaoCrediario

                            For Each row In dt.Rows
                                item = New dCrediario

                                item.cid = cFuncoes.RetornarInteiro(row("cid"))
                                item.controle = cFuncoes.RetornarInteiro(row("controle"))
                                item.usuarioId = cFuncoes.RetornarInteiro(row("usuarioId"))
                                item.clienteId = cFuncoes.RetornarInteiro(row("clienteId"))
                                item.Parcelas = cFuncoes.RetornarInteiro(row("parcelas"))
                                item.ValorTotal = cFuncoes.RetornarDecimal(row("valortotal"))
                                item.ValorPago = cFuncoes.RetornarDecimal(row("valorpago"))
                                item.SaldoDevedor = cFuncoes.RetornarDecimal(row("saldodevedor"))
                                item.DataVenda = cFuncoes.RetornarData(row("dataVenda"))

                                retorno.Add(item)
                            Next
                        Else
                            retorno = Nothing
                        End If
                    Else
                        retorno = Nothing
                    End If
                Else
                    retorno = Nothing
                End If

            Catch ex As Exception

                retorno = Nothing
                Throw New ExcecaoNascomercio("Erro em Consultar Crediario[" & Me.ToString() & "] - " & ex.Message)

            End Try

            Return retorno

        End Function

        Public Function Incluir(ByVal dados As dCrediario) As Integer

            Dim retorno As Integer
            Dim acessoBanco As cAcessoBD
            Dim comandoSQL As String


            Try

                acessoBanco = New cAcessoBD


                comandoSQL = " INSERT INTO " & _
                    " crediario (controle, usuarioId, clienteId, observacao, " & _
                             "parcelas, valortotal, valorpago, saldodevedor, dataVenda, loja_cid, terminal) " & _
                    " VALUES (" & _
                            cFuncoes.PersistirTexto(dados.controle) & "," & _
                            cFuncoes.PersistirTexto(dados.usuarioId) & "," & _
                            cFuncoes.PersistirTexto(dados.clienteId) & "," & _
                            cFuncoes.PersistirTexto(dados.NotaFiscal) & "," & _
                            cFuncoes.PersistirInteiro(dados.Parcelas) & "," & _
                            cFuncoes.PersistirDecimal(dados.ValorTotal) & "," & _
                            cFuncoes.PersistirDecimal(dados.ValorPago) & "," & _
                            cFuncoes.PersistirDecimal(dados.SaldoDevedor) & "," & _
                            cFuncoes.PersistirData(dados.DataVenda) & "," & _
                            cFuncoes.PersistirInteiro(dados.LojaId) & "," & _
                            cFuncoes.PersistirTexto(dados.Terminal) & ")"

                retorno = acessoBanco.ExecutarCID(comandoSQL)

            Catch ex As Exception

                retorno = Nothing
                Throw New ExcecaoNascomercio("Erro em Incluir Crediario [" & Me.ToString() & "] - " & ex.Message)

            End Try

            Incluir = retorno

        End Function

        Public Function Alterar(ByVal dados As dCrediario) As Integer

            Dim retorno As Integer
            Dim acessoBanco As cAcessoBD
            Dim comandoSQL As String

            Try

                acessoBanco = New cAcessoBD

                comandoSQL = " UPDATE crediario SET " & _
                    " controle = " & cFuncoes.PersistirInteiro(dados.controle) & "," & _
                    " usuarioId = " & cFuncoes.PersistirTexto(dados.usuarioId) & "," & _
                    " clienteId = " & cFuncoes.PersistirTexto(dados.clienteId) & "," & _
                    " parcelas = " & cFuncoes.PersistirInteiro(dados.Parcelas) & "," & _
                    " valortotal = " & cFuncoes.PersistirDecimal(dados.ValorTotal) & "," & _
                    " valorpago = " & cFuncoes.PersistirDecimal(dados.ValorPago) & "," & _
                    " dataVenda = " & cFuncoes.PersistirData(dados.DataVenda) & "," & _
                    " saldodevedor = " & cFuncoes.PersistirDecimal(dados.SaldoDevedor) & _
                    " WHERE " & _
                    " cid = " & dados.cid.ToString()

                retorno = acessoBanco.ExecutarINT(comandoSQL)

            Catch ex As Exception

                retorno = Nothing
                Throw New ExcecaoNascomercio("Erro em Alterar crediario [" & Me.ToString() & "] - " & ex.Message)

            End Try

            Alterar = retorno

        End Function

        Public Function AlterarControle(ByVal dados As dCrediario) As Integer

            Dim retorno As Integer
            Dim acessoBanco As cAcessoBD
            Dim comandoSQL As String

            Try

                acessoBanco = New cAcessoBD

                comandoSQL = " UPDATE crediario SET " & _
                    " controle = " & cFuncoes.PersistirInteiro(dados.controle) & _
                    " WHERE " & _
                    " controle = " & dados.cid.ToString()

                retorno = acessoBanco.ExecutarINT(comandoSQL)

            Catch ex As Exception

                retorno = Nothing
                Throw New ExcecaoNascomercio("Erro em Alterar crediario [" & Me.ToString() & "] - " & ex.Message)

            End Try

            Return retorno

        End Function

        Public Function Excluir(ByVal dados As dCrediario) As Integer

            Dim retorno As Integer
            Dim acessoBanco As cAcessoBD
            Dim comandoSQL As String

            Try

                acessoBanco = New cAcessoBD

                comandoSQL = " DELETE FROM crediario " & _
                    " WHERE cid = " & dados.cid.ToString()

                retorno = acessoBanco.ExecutarINT(comandoSQL)

            Catch ex As Exception

                retorno = Nothing
                Throw New ExcecaoNascomercio("Erro em Excluir crediario [" & Me.ToString() & "] - " & ex.Message)

            End Try

            Excluir = retorno

        End Function


        Public Function ExcluirParcelas(ByVal dados As dCrediario) As Integer

            Dim retorno As Integer
            Dim acessoBanco As cAcessoBD
            Dim comandoSQL As String

            Try

                acessoBanco = New cAcessoBD

                comandoSQL = " DELETE FROM parcelas " & _
                    " WHERE crediarioid = " & dados.cid.ToString()

                retorno = acessoBanco.ExecutarINT(comandoSQL)

            Catch ex As Exception

                retorno = Nothing
                Throw New ExcecaoNascomercio("Erro em Excluir crediario [" & Me.ToString() & "] - " & ex.Message)

            End Try

            Return retorno

        End Function

    End Class


End Namespace

File: ncRepositorios/pCrediario.cs
using System;
using System.Data;
using ncNComum.nsAcessoBD;
using ncNComum.nsExcecao;
using static ncNComum.nsFuncoes.cFuncoes;
using nsCrediario;

namespace ncPersistencia.nsCrediario
{
    public class pCrediario
    {
        public ColecaoCrediario Listar()
        {
            ColecaoCrediario retorno = null;
            try
            {
                var acessoBanco = new cAcessoBD();
                string comandoSQL = " Select cid, controle, usuarioId, clienteId, " +
                    "parcelas, valortotal, valorpago, saldodevedor, dataVenda From crediario";
                var ds = acessoBanco.ExecutarDS(comandoSQL);
                if (ds != null && ds.Tables.Count > 0)
                {
                    DataTable dt = ds.Tables[0];
                    if (dt.Rows.Count > 0)
                    {
                        retorno = new ColecaoCrediario();
                        foreach (DataRow row in dt.Rows)
                        {
                            var item = new dCrediario();
                            item.cid = RetornarInteiro(row["cid"]);
                            item.controle = RetornarInteiro(row["controle"]);
                            item.ValorTotal = PersistirDecimal(row["valortotal"]);
                            item.ValorTotal = PersistirDecimal(row["valorpago"]);
                            item.ValorTotal = PersistirDecimal(row["saldodevedor"]);
                            item.usuarioId = RetornarInteiro(row["usuarioId"]);
                            item.clienteId = RetornarInteiro(row["clienteId"]);
                            item.Parcelas = PersistirInteiro(row["parcelas"]);
                            item.DataVenda = RetornarData(row["dataVenda"]);
                            retorno.Add(item);
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                retorno = null;
                throw new ExcecaoNascomercio("Erro em Listar Crediário [" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }

        public int ConsultarMax()
        {
            int retorno = 0;
            try
            {
                var acessoBanco = new cAcessoBD();
                string sqlSelect = " Select MAX(controle) as controle";
                string sqlFrom = " From crediario ";
                var ds = acessoBanco.ExecutarDS(sqlSelect + " " + sqlFrom);
                if (ds != null && ds.Tables.Count > 0)
                {
                    DataTable dt = ds.Tables[0];
                    if (dt.Rows.Count > 0)
                    {
                        foreach (DataRow row in dt.Rows)
                            retorno = row["controle"] == DBNull.Value ? 0 : RetornarInteiro(row["controle"]);
                    }
                }
            }
            catch (Exception ex)
            {
                retorno = 0;
                throw new ExcecaoNascomercio("Erro em ConsultarMax Venda [" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }

        public ColecaoCrediario Consultar(dCrediario dados)
        {
            ColecaoCrediario retorno = null;
            try
            {
                var acessoBanco = new cAcessoBD();
                string sqlSelect = " Select cid, controle, usuarioId, clienteId, " +
                    "parcelas, valortotal, valorpago, saldodevedor, dataVenda ";
                string sqlWhere = string.Empty;
                string sqlFrom = " From crediario ";
                if (dados.cid != 0)
                    sqlWhere = MontarParametrosSQL(sqlWhere, dados.cid, "cid");
                if (dados.controle != 0)
                    sqlWhere = MontarParametrosSQL(sqlWhere, dados.controle, "controle");
                if (dados.usuarioId != 0)
                    sqlWhere = MontarParametrosSQL(sqlWhere, dados.usuarioId, "usuarioId");
                if (dados.clienteId != 0)
                    sqlWhere = MontarParametrosSQL(sqlWhere, dados.clienteId, "clienteId");
                if (dados.SaldoDevedor > 0.001m)
                {
                    if (!sqlWhere.Equals(string.Empty))
                        sqlWhere += " AND saldodevedor > 0.001 ";
                    else
                        sqlWhere = " saldodevedor > 0.001 ";
                }
                if (!sqlWhere.Equals(string.Empty))
                    sqlWhere = " WHERE " + sqlWhere;
                var ds = acessoBanco.ExecutarDS(sqlSelect + " " + sqlFrom + " " + sqlWhere + " order by year(dataVenda), month(dataVenda)");
                if (ds != null && ds.Tables.Count > 0)
                {
                    DataTable dt = ds.Tables[0];
                    if (dt.Rows.Count > 0)
                    {
                        retorno = new ColecaoCrediario();
                        foreach (DataRow row in dt.Rows)
                        {
                            var item = new dCrediario();
                            item.cid = RetornarInteiro(row["cid"]);
                            item.controle = RetornarInteiro(row["controle"]);
                            item.usuarioId = RetornarInteiro(row["usuarioId"]);
                            item.clienteId = RetornarInteiro(row["clienteId"]);
                            item.Parcelas = RetornarInteiro(row["parcelas"]);
                            item.ValorTotal = RetornarDecimal(row["valortotal"]);
                            item.ValorPago = RetornarDecimal(row["valorpago"]);
                            item.SaldoDevedor = RetornarDecimal(row["saldodevedor"]);
                            item.DataVenda = RetornarData(row["dataVenda"]);
                            retorno.Add(item);
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                retorno = null;
                throw new ExcecaoNascomercio("Erro em Consultar Crediario[" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }

        public int Incluir(dCrediario dados)
        {
            int retorno = 0;
            try
            {
                var acessoBanco = new cAcessoBD();
                string comandoSQL = " INSERT INTO " +
                    " crediario (controle, usuarioId, clienteId, observacao, " +
                    "parcelas, valortotal, valorpago, saldodevedor, dataVenda, loja_cid, terminal) " +
                    " VALUES (" +
                    PersistirTexto(dados.controle) + "," +
                    PersistirTexto(dados.usuarioId) + "," +
                    PersistirTexto(dados.clienteId) + "," +
                    PersistirTexto(dados.NotaFiscal) + "," +
                    PersistirInteiro(dados.Parcelas) + "," +
                    PersistirDecimal(dados.ValorTotal) + "," +
                    PersistirDecimal(dados.ValorPago) + "," +
                    PersistirDecimal(dados.SaldoDevedor) + "," +
                    PersistirData(dados.DataVenda) + "," +
                    PersistirInteiro(dados.LojaId) + "," +
                    PersistirTexto(dados.Terminal) + ")";
                retorno = acessoBanco.ExecutarCID(comandoSQL);
            }
            catch (Exception ex)
            {
                retorno = 0;
                throw new ExcecaoNascomercio("Erro em Incluir Crediario [" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }

        public int Alterar(dCrediario dados)
        {
            int retorno = 0;
            try
            {
                var acessoBanco = new cAcessoBD();
                string comandoSQL = " UPDATE crediario SET " +
                    " controle = " + PersistirInteiro(dados.controle) + "," +
                    " usuarioId = " + PersistirTexto(dados.usuarioId) + "," +
                    " clienteId = " + PersistirTexto(dados.clienteId) + "," +
                    " parcelas = " + PersistirInteiro(dados.Parcelas) + "," +
                    " valortotal = " + PersistirDecimal(dados.ValorTotal) + "," +
                    " valorpago = " + PersistirDecimal(dados.ValorPago) + "," +
                    " dataVenda = " + PersistirData(dados.DataVenda) + "," +
                    " saldodevedor = " + PersistirDecimal(dados.SaldoDevedor) +
                    " WHERE " +
                    " cid = " + dados.cid.ToString();
                retorno = acessoBanco.ExecutarINT(comandoSQL);
            }
            catch (Exception ex)
            {
                retorno = 0;
                throw new ExcecaoNascomercio("Erro em Alterar crediario [" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }

        public int AlterarControle(dCrediario dados)
        {
            int retorno = 0;
            try
            {
                var acessoBanco = new cAcessoBD();
                string comandoSQL = " UPDATE crediario SET " +
                    " controle = " + PersistirInteiro(dados.controle) +
                    " WHERE " +
                    " controle = " + dados.cid.ToString();
                retorno = acessoBanco.ExecutarINT(comandoSQL);
            }
            catch (Exception ex)
            {
                retorno = 0;
                throw new ExcecaoNascomercio("Erro em Alterar crediario [" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }

        public int Excluir(dCrediario dados)
        {
            int retorno = 0;
            try
            {
                var acessoBanco = new cAcessoBD();
                string comandoSQL = " DELETE FROM crediario " +
                    " WHERE cid = " + dados.cid.ToString();
                retorno = acessoBanco.ExecutarINT(comandoSQL);
            }
            catch (Exception ex)
            {
                retorno = 0;
                throw new ExcecaoNascomercio("Erro em Excluir crediario [" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }

        public int ExcluirParcelas(dCrediario dados)
        {
            int retorno = 0;
            try
            {
                var acessoBanco = new cAcessoBD();
                string comandoSQL = " DELETE FROM parcelas " +
                    " WHERE crediarioid = " + dados.cid.ToString();
                retorno = acessoBanco.ExecutarINT(comandoSQL);
            }
            catch (Exception ex)
            {
                retorno = 0;
                throw new ExcecaoNascomercio("Erro em Excluir crediario [" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }
    }
}
